packages/tools/pactjs-cli/src/contract-generate/generate.ts

```diff
@@ -47,145 +48,213 @@ function verifyTsconfigTypings(
+    const [contract] = args.contract;
 
-      // if contract is namespaced, use the first part as the namespace
-      const namespace = args.contract.includes('.')
-        ? args.contract.split('.')[0]
-        : undefined;
+    const pactCode = await retrieveContractFromChain(
+      contract,
+      args.api!,
+      args.chain!,
+      args.network!,
+    );
 
-      pactModule = new StringContractDefinition({
-        contract: pactCode,
-        namespace,
-      });
-    } else {
-      console.log(`Generating Pact contract from ${args.file}`);
-      pactModule = new FileContractDefinition({
-        path: join(process.cwd(), args.file!),
-      });
+    if (pactCode === undefined || pactCode.length === 0) {
+      program.error('Could not retrieve contract from chain');
     }
 
-    const moduleDtss: Map<string, string> = generateDts(
-      pactModule.modulesWithFunctions,
-      args.capsInterface,
+    // if contract is namespaced, use the first part as the namespace
+    const namespace = contract.includes('.')
+      ? contract.split('.')[0]
+      : undefined;
+
+    pactModule = new StringContractDefinition({
+      contract: pactCode,
+      namespace,
+    });
+  } else {
+    const [file] = args.file!;
+    console.log(`Generating Pact contract from ${file}`);
+    pactModule = new FileContractDefinition({
+      path: join(process.cwd(), file!),
+    });
+  }
+
+  const moduleDtss: Map<string, string> = generateDts(
+    pactModule.modulesWithFunctions,
+    args.capsInterface,
+  );
+  return moduleDtss;
+}
+
+async function generatorV2(
+  args: IContractGenerateOptions,
+): Promise<Map<string, string>> {
+  if (args.contract !== undefined) {
+    console.log(
+      `Generating pact contracts from chainweb for ${args.contract.join(',')}`,
     );
+  }
 
-    // walk up in file tree from process.cwd() to get the package.json
-    const targetPackageJson: string | undefined = shallowFindFile(
-      process.cwd(),
-      'package.json',
+  if (args.file !== undefined) {
+    console.log(
+      `Generating pact contracts from files for ${args.file.join(',')}`,
     );
+  }
 
+  const getContract = async (name: string): Promise<string> => {
+    console.log('fetching', name);
     if (
-      targetPackageJson === undefined ||
-      targetPackageJson.length === 0 ||
-      targetPackageJson === '/'
+      args.api !== undefined &&
+      args.chain !== undefined &&
+      args.network !== undefined
     ) {
-      program.error('Could not find package.json');
+      const content = await retrieveContractFromChain(
+        name,
+        args.api,
+        args.chain,
+        args.network,
+      );
+      return content ?? '';
     }
+    console.log(`
+      the generator tries to fetch ${name} from the blockchain but the api data is not presented.
+      this happen because ${name} mentioned via --contracts directly or it is a dependency of a module.
+      the scrip skips this module
+      `);
+    return '';
+  };
```

Comment:
Do we want to log with the indents? This log means that it'll be written like so

```sh
$ some command
      the generator tries to fetch ${name} from the blockchain but the api data is not presented.
      this happen because ${name} mentioned via --contracts directly or it is a dependency of a module.
      the scrip skips this module
```

Otherwise we can do this:
```suggestion
    console.log(`
the generator tries to fetch ${name} from the blockchain but the api data is not presented.
this happen because ${name} mentioned via --contracts directly or it is a dependency of a module.
the scrip skips this module\n`);
    return '';
  };
```